src/pipeline.cc

```diff
@@ -1219,7 +1271,12 @@ NAN_METHOD(pipeline) {
 
   // Join queue for worker thread
   Callback *callback = new Callback(info[1].As<Function>());
-  AsyncQueueWorker(new PipelineWorker(callback, baton, queueListener, bufferIn, overlayBufferIn));
+
+  std::vector<BufferContainer> saveBuffers;
+  saveBuffers.push_back({"bufferIn", bufferIn, baton->bufferInLength});
```

Comment:
If we only append buffers to the vector where their length is >0 then we don't need to store the length.